### DISCUSSION TITLE
What is the best practice when creating tables?

### DISCUSSION THREAD

**Discussion by kyto64** [Dynamoid/dynamoid#690] [Q&A] [answered]
In my service, when a value does not exist in a table, I store that value.

When traversing the table, if the table does not exist, of course `Aws::DynamoDB::Errors::ResourceNotFoundException` is raised.
Is it a best practice to create the table in advance, e.g. rake dynamoid:create_tables, to prevent this expected exception from occurring?

Note that I do not want to check for the existence of the table each time the table is traversed in logic, for performance reasons.

Any ideas would be appreciated.
Thanks.

**Comment by andrykonchin** [comment, marked as answer]
Tables creation is kind of database management area that IMHO is not a responsibility or purpose of an ORM. AFAIK a table creation on the fly is just a handy way to initialise a database or table in the test environment.

So it isn't a common practice AFAIK to create tables using Dynamoid in production environment.
